next-unbuilt-package.sh

```diff
@@ -48,10 +49,14 @@ for CRATE in $POPULAR_CRATES; do
     continue
   fi
 
-  rm -rf "$TEMPDIR/crates.io-response.json"
-  curl_slowly --location --fail "https://crates.io/api/v1/crates/${CRATE}" >"$TEMPDIR/crates.io-response.json"
-  VERSION=$(cat "$TEMPDIR/crates.io-response.json" | jq -r .versions[0].num)
-  LICENSE=$(cat "$TEMPDIR/crates.io-response.json" | jq -r .versions[0].license | sed -e 's:/:", ":g' -e 's/ OR /", "/g')
+  RESPONSE_DIR="$TEMPDIR/crates.io-responses/"
+  RESPONSE_FILENAME="$RESPONSE_DIR/$CRATE.json"
+  if [[ ! -f "$RESPONSE_FILENAME" ]]; then
+    mkdir -p "$RESPONSE_DIR"
+    curl_slowly --location --fail "https://crates.io/api/v1/crates/${CRATE}" >"$RESPONSE_FILENAME"
+  fi
+  VERSION=$(cat "$RESPONSE_FILENAME" | jq -r .versions[0].num)
```

Comment:
```suggestion
  VERSION=$(jq -r '.versions[0].num' "$RESPONSE_FILENAME")
```

Reply:
You will notice that I tend to use `cat | ...` a lot. I find that it reduces cognitive load, because tools are only used in their "filter" mode, and information flow is more uniformly left to right throughout the codebase. 

As an added bonus, `cat | jq` is faster for prototyping on the command line, because you are always editing the last word on the line each time you iterate.

Shellcheck also dislikes `cat | ...`, so I am probably alone in my preference. I will do a pass to make things more best-practice compliant after breakfast.